tools/ctl/ctl.c

```diff
@@ -355,12 +356,13 @@ static int ctl_setup(struct ctl_data *ctl_data)
 	read = snd_ctl_elem_info_is_tlv_readable(ctl_data->info);
 	write = snd_ctl_elem_info_is_tlv_writable(ctl_data->info);
 	type = snd_ctl_elem_info_get_type(ctl_data->info);
+	is_volatile = snd_ctl_elem_info_is_volatile(ctl_data->info);
 	if (!read) {
 		fprintf(stderr, "Error: No read capability.\n");
 		goto value_free;
 	}
-	if (!write) {
-		fprintf(stderr, "Error: No write capability.\n");
+	if (!write && !(read & is_volatile)) {
+		fprintf(stderr, "Error: No write capability or Not a read-only control.\n");
```

Comment:
This is not easy for the developer to debug with the "or". Can we have an unique error message for each failure.

Reply:
also a readwrite control can be volatile as well. So the best check for readonly is (read && !write)

Reply:
Done